mono/metadata/threads.c

```diff
@@ -2012,11 +2015,13 @@ gint32 ves_icall_System_Threading_WaitHandle_WaitOne_internal(HANDLE handle, gin
 	if (mono_thread_current_check_pending_interrupt ())
 		return map_native_wait_result_to_managed (MONO_W32HANDLE_WAIT_RET_FAILED, 0);
 
+	mono_w32handle_ref(handle);
```

Comment:
What do you mean by "checking if it's valid" here that isn't covered by `mono_w32handle_ref` ? 

`mono_w32handle_lock_handle` itself calls `mono_w32handle_ref` before trying on the lock too, so "ref before lock" is the right order. The reason I moved the inc here for these functions is because they were removed from `mono_join_uninterrupted`.